src/Build/BackEnd/Components/RequestBuilder/TaskHost.cs

```diff
@@ -908,13 +908,23 @@ internal void ReleaseAllCores()
         [Serializable]
         private sealed class EngineServicesImpl : EngineServices
         {
+            /// <summary>
+            /// Adds <see cref="RequestEntryProjectFile"/>.
+            /// </summary>
+            private const int Version2 = 2;
+
             private readonly TaskHost _taskHost;
 
             internal EngineServicesImpl(TaskHost taskHost)
             {
                 _taskHost = taskHost;
             }
 
+            public override int Version => Version2;
+
+            /// <inheritdoc/>
+            public override string RequestEntryProjectFile => _taskHost._requestEntry.RequestConfiguration.ProjectFullPath;
```

Comment:
What's this change for?

Reply:
Determined offline that this can be obtainable with `$(MSBuildProjectFullPath)`.